src/gt4py/next/type_system_2/traits.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+import abc
+from typing import Any, Optional
+from collections.abc import Sequence
+
+import dataclasses
+
+
+class Trait:
+    ...
+
+
+class SignednessTrait(Trait):
```

Comment:
Not sure, but maybe `Protocol`s could be more appropriate for this pattern.

Reply:
Do you have a minimal example of how you think we could use protocols, or any other constructs, for that matter? I haven't spent much time exploring other options but I can imagine another construct could be a better fit.